Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Extensions/ApiGatewayResponseExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace Amazon.Lambda.TestTool.Extensions;
+
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.IO;
+using System.Text;
+
+/// <summary>
+/// Provides extension methods for converting API Gateway responses to HttpResponse objects.
+/// </summary>
+public static class ApiGatewayResponseExtensions
+{
+    /// <summary>
+    /// Converts an APIGatewayProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway proxy response to convert.</param>
+    /// <returns>An HttpResponse object representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayProxyResponse apiResponse)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        response.StatusCode = apiResponse.StatusCode;
+
+        SetResponseHeaders(response, apiResponse.Headers, apiResponse.MultiValueHeaders);
+
+        SetResponseBody(response, apiResponse.Body, apiResponse.IsBase64Encoded);
+
+        return response;
+    }
+
+    /// <summary>
+    /// Converts an APIGatewayHttpApiV2ProxyResponse to an HttpResponse.
+    /// </summary>
+    /// <param name="apiResponse">The API Gateway HTTP API v2 proxy response to convert.</param>
+    /// <returns>An HttpResponse object representing the API Gateway response.</returns>
+    public static HttpResponse ToHttpResponse(this APIGatewayHttpApiV2ProxyResponse apiResponse)
+    {
+        var httpContext = new DefaultHttpContext();
+        var response = httpContext.Response;
+
+        response.StatusCode = apiResponse.StatusCode;
+
+        SetResponseHeaders(response, apiResponse.Headers);
+
+        if (apiResponse.Cookies != null)
+        {
+            foreach (var cookie in apiResponse.Cookies)
+            {
+                response.Headers.Append("Set-Cookie", cookie);
+            }
+        }
+
+        SetResponseBody(response, apiResponse.Body, apiResponse.IsBase64Encoded);
+
+        return response;
+    }
+
+    /// <summary>
+    /// Sets the headers on the HttpResponse object.
+    /// </summary>
+    /// <param name="response">The HttpResponse object to modify.</param>
+    /// <param name="headers">The single-value headers to set.</param>
+    /// <param name="multiValueHeaders">The multi-value headers to set.</param>
+    private static void SetResponseHeaders(HttpResponse response, IDictionary<string, string>? headers, IDictionary<string, IList<string>>? multiValueHeaders = null)
+    {
+        if (headers != null)
+        {
+            foreach (var header in headers)
+            {
+                response.Headers[header.Key] = header.Value;
```

Comment:
yeah you are right. i just checked by making a function that returns

```
return {
        statusCode: 200,
        headers: {
            'Content-Type': 'application/json',
            'myheader': 'test,shouldhavesecondvalue',
        },
```

and _curling_ it. and the response header only shows `myheader: shouldhavesecondvalue`. in my current code, i beleive it would've been  'test,shouldhavesecondvalue which is not correct. It should be the last header value for that key

Reply:
lambda code

```
 return {
        statusCode: 200,
        headers: {
            'Content-Type': 'application/json',
            'myheader': 'test,shouldhavesecondvalue',
            'anotherheader': 'firstvalue',
            'anotherheader': 'secondvalue'
        },
        multiValueHeaders: {
            "headername": ["headervalue", "headervalue2"]
        },
        body: JSON.stringify({
            event
        }),    };
};
```

V1 Response headers
```
Myheader: test,shouldhavesecondvalue
Anotherheader: secondvalue
Headername: headervalue
Headername: headervalue2
```

v2 response headers

```
Anotherheader: secondvalue
Myheader: test,shouldhavesecondvalue
```

so i think my code is correct then no?